src/main/resources/com/tresys/nitf/xsd/nitf_extension_types.dfdl.xsd

```diff
@@ -0,0 +1,148 @@
+    this list of conditions and the following disclaimers.
+
+ 2. Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimers in the
+    documentation and/or other materials provided with the distribution.
+
+ 3. Neither the names of Tresys Technology, nor the names of its contributors
+    may be used to endorse or promote products derived from this Software
+    without specific prior written permission.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS WITH THE
+SOFTWARE.
+-->
+
+
+<!--
+Version 0.0.1  nitf_extension_types.dfdl.xsd
+
+This DFDL schema provides basic, default support for various extendable types
+in NITF, such as TaggedRecordExtensions and ImageData formats.
+-->
+
+<xs:schema targetNamespace="urn:nitfExtensionTypes" xmlns:tns="urn:nitfExtensionTypes"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:dfdlx="http://www.ogf.org/dfdl/dfdl-1.0/extensions"
+  xmlns:nitf="urn:nitf:2.1"
+  xmlns:nitfc="urn:nitfCommonTypes"
+  elementFormDefault="unqualified">
+
+  <xs:import namespace="urn:nitfCommonTypes" schemaLocation="com/tresys/nitf/xsd/nitf_common_types.dfdl.xsd" />
+  <xs:import namespace="urn:nitf:2.1" schemaLocation="com/tresys/nitf/xsd/nitf.dfdl.xsd" />
+
+  <xs:import schemaLocation="com/mitre/jpeg/xsd/jpeg.dfdl.xsd" />
+
+  <xs:annotation>
+    <xs:appinfo source="http://www.ogf.org/dfdl/">
+      <dfdl:format ref="nitfc:nitfGeneralFormat" />
+    </xs:appinfo>
+  </xs:annotation>
+
+  <xs:complexType name="taggedRecordExtension">
+    <xs:sequence>
+      <xs:element name="UniqueExtensionTypeIdentifier" type="nitfc:BCS-A" dfdl:length="6" />
+      <xs:element name="LengthOfData" type="nitfc:BCS-NP" dfdl:length="5" dfdl:outputValueCalc="{ dfdl:valueLength(../UserDefinedData, 'bytes') }" />
+      <xs:sequence>
+        <xs:annotation>
+          <xs:appinfo source="http://www.ogf.org/dfdl/">
+            <dfdl:newVariableInstance ref="nitf:payloadSizeInBytes" />
+            <dfdl:setVariable ref="nitf:payloadSizeInBytes" value="{ LengthOfData }" />
+          </xs:appinfo>
+        </xs:annotation>
+        <xs:element name="UserDefinedData" type="nitfc:hexBinaryOrBlob" dfdl:lengthKind="explicit" dfdl:length="{ ../LengthOfData }" />
+      </xs:sequence>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="imageData">
```

Comment:
Can some parts of this be made into reusable groups that are defined in the main nitf schema?

Ex: can the entire complex type of MaskTable be factored out as MaskTableType? 

The Choice below at line 132 I can see has to be in this file. But the rest doesn't seem to depend on details within the choice. 

If I'm wrong and this MaskTable does depend on the JPEG behavior, then I'd still suggest pulling out the type as a named complex type just so that type can be named JPEGMaskTable, or MaskTableForJPEGImageData or something like that to make clear it is specific.

Reply:
Makes sense, if some of this is the same as raster, it could be pulled out into the nitif common and shared between the two.

Reply:
I have moved it to its own type now.  There is some overlap with the MaskTable that is defined in Raster, but they call certain elements different names in Raster and other parts are completely different.